packages/plugin-workspaces-tools/sources/commands/foreach.ts

```diff
@@ -0,0 +1,128 @@
+import { Configuration, IdentHash, PluginConfiguration, Project, Workspace } from '@berry/core';
+import { structUtils, Cache, DescriptorHash, StreamReport }                  from '@berry/core';
+import { Writable }                                                          from 'stream';
+import { cpus }                                                              from 'os';
+// @ts-ignore
+import pLimit                                                                from 'p-limit';
+
+import * as workspaceUtils                                                   from '../workspaceUtils';
+
+type ForeachOptions = {
+  args: Array<string>;
+  command: string;
+  cwd: string;
+  exclude: string[];
+  include: string[];
+  interlaced: boolean;
+  parallel: boolean;
+  prefixed: boolean;
+  stdout: Writable;
+  withDependencies: boolean;
+}
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: any, pluginConfiguration: PluginConfiguration) => clipanion
+
+    .command(`workspaces foreach <command> [...args] [-p,--parallel] [--with-dependencies] [-I,--interlaced] [-P,--prefixed] [-i,--include WORKSPACES...] [-x,--exclude WORKSPACES...]`)
+    .flags({proxyArguments: true})
+
+    .categorize(`Workspace-related commands`)
+    .describe(`run a command on all workspaces`)
+
+    .action(
+      async ({cwd, args, stdout, command, exclude, include, interlaced, parallel, withDependencies, prefixed, ...env}: ForeachOptions) => {
+        const configuration = await Configuration.find(cwd, pluginConfiguration);
+        const { project } = await Project.find(configuration, cwd);
+        const cache = await Cache.find(configuration);
+
+        const needsProcessing = new Map<IdentHash, Workspace>();
+        const processing = new Set<DescriptorHash>();
+        const concurrency = parallel ? Math.max(1, cpus().length / 2) : 1;
+        const limit = pLimit(concurrency);
+        let commandCount = 0;
+
+        const report = await StreamReport.start({configuration, stdout}, async report => {
+          await project.resolveEverything({lockfileOnly: true, cache, report});
+
+          let workspaces = command.toLocaleLowerCase() === `run`
```

Comment:
I'd use `toLowerCase` since we're working with ascii here

```suggestion
          let workspaces = command.toLowerCase() === `run`
```